pkgs/by-name/op/openvpn3/package.nix

```
@@ -0,0 +1,133 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, asio
+, glib
+, jsoncpp
+, libcap_ng
+, libnl
+, libuuid
+, lz4
+, openssl
+, pkg-config
+, protobuf
+, python3
+, systemd
+, tinyxml-2
+, wrapGAppsHook3
+, gobject-introspection
+, meson
+, ninja
+, gdbuspp
+, cmake
+, git
+, enableSystemdResolved ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openvpn3";
+  # also update openvpn3-core
+  version = "22_dev";
+
+  src = fetchFromGitHub {
+    owner = "OpenVPN";
+    repo = "openvpn3-linux";
+    rev = "refs/tags/v${version}";
+    # Don't forget to actualize version scripts in `postPatch`
+    hash = "sha256-UbphN5gHgO30ry0kX6W4qSs5Ksrvfhm4xiRNBdzEOhA=";
+    # `openvpn3-core` is a submodule.
+    # TODO: make it into a separate package
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    ./patches/0001-customizable-asio-path.patch
+    ./patches/0002-customizable-installation-paths.patch
+  ];
+
+  postPatch = ''
+    echo '#define OPENVPN_VERSION "3.git:unknown:unknown"' > ./src/build-version.h
+    echo '#define PACKAGE_GUIVERSION "${builtins.replaceStrings ["_"] [":"] version}_unknown__s"' >> ./src/build-version.h
+    echo '#define PACKAGE_NAME "openvpn3-linux"' >> ./src/build-version.h
+
+    patchShebangs ** /*.py ** /*.sh \
+      ./scripts \
+      ./src/python/{openvpn2,openvpn3-as,openvpn3-autoload} \
+      ./distro/systemd/openvpn3-systemd \
+      ./src/tests/dbus/netcfg-subscription-test
+  '';
+
+  pythonPath = python3.withPackages (ps: [
+    ps.dbus-python
+    ps.pygobject3
+    ps.systemd
+  ]);
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    cmake
+    git
+
+    python3.pkgs.wrapPython
+    python3.pkgs.docutils
+    python3.pkgs.jinja2
+    python3.pkgs.dbus-python
+    wrapGAppsHook3
+    gobject-introspection.dev
```

```suggestion
    gobject-introspection
```
this is likely wrong or in the wrong input

---

Removed `.dev`, yet it is consciously on `nativeBuildInputs`